firmware/makefile

```diff
@@ -1,5 +1,7 @@
 # Makefile by Adam, 2019
-
+include user.cfg
```

Comment:
Why is this a separate file? If it's checked into the repo (which it is), keeping it in one makefile is cleaner and simpler.
If this is supposed to be configurable per-user, then it shouldn't be a dependency or checked into the repo. It's better to check an environment variable in that case, like `ESP_PORT`

Reply:
Most of the gdb stuff came from Richard. Originally to use gdb there were two lines that needed to commented out, one in the makefile and one somewhere else. I added this user.cfg to have one place to get this option. Having to modify the makefile didnt seem like a good idea. You are right user.cfg shouldnt be in the repo. I not present gdb will be off.

Reply:
I changed it in my master so that the makefile checks if the environment variable `ESP_GDB` exists.